vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataType.java

```diff
@@ -46,85 +47,85 @@
-  FLOAT8(701, true, Double.class, Number.class),
-  FLOAT8_ARRAY(1022, true, Double[].class, Number[].class),
-  NUMERIC(1700, false, Numeric.class, Number.class),
-  NUMERIC_ARRAY(1231, false, Numeric[].class, Number[].class),
-  MONEY(790, true, Object.class),
-  MONEY_ARRAY(791, true, Object[].class),
-  BIT(1560, true, Object.class),
-  BIT_ARRAY(1561, true, Object[].class),
-  VARBIT(1562, true, Object.class),
-  VARBIT_ARRAY(1563, true, Object[].class),
-  CHAR(18, true, String.class),
-  CHAR_ARRAY(1002, true, String[].class),
-  VARCHAR(1043, true, String.class),
-  VARCHAR_ARRAY(1015, true, String[].class),
-  BPCHAR(1042, true, String.class),
-  BPCHAR_ARRAY(1014, true, String[].class),
-  TEXT(25, true, String.class),
-  TEXT_ARRAY(1009, true, String[].class),
-  NAME(19, true, String.class),
-  NAME_ARRAY(1003, true, String[].class),
-  DATE(1082, true, LocalDate.class),
-  DATE_ARRAY(1182, true, LocalDate[].class),
-  TIME(1083, true, LocalTime.class),
-  TIME_ARRAY(1183, true, LocalTime[].class),
-  TIMETZ(1266, true, OffsetTime.class),
-  TIMETZ_ARRAY(1270, true, OffsetTime[].class),
-  TIMESTAMP(1114, true, LocalDateTime.class),
-  TIMESTAMP_ARRAY(1115, true, LocalDateTime[].class),
-  TIMESTAMPTZ(1184, true, OffsetDateTime.class),
-  TIMESTAMPTZ_ARRAY(1185, true, OffsetDateTime[].class),
-  INTERVAL(1186, true, Interval.class),
-  INTERVAL_ARRAY(1187, true, Interval[].class),
-  BYTEA(17, true, Buffer.class),
-  BYTEA_ARRAY(1001, true, Buffer[].class),
-  MACADDR(829, true, Object.class),
-  INET(869, true, Object[].class),
-  CIDR(650, true, Object.class),
-  MACADDR8(774, true, Object[].class),
-  UUID(2950, true, UUID.class),
-  UUID_ARRAY(2951, true, UUID[].class),
-  JSON(114, true, Object.class),
-  JSON_ARRAY(199, true, Object[].class),
-  JSONB(3802, true, Object.class),
-  JSONB_ARRAY(3807, true, Object[].class),
-  XML(142, true, Object.class),
-  XML_ARRAY(143, true, Object[].class),
-  POINT(600, true, Point.class),
-  POINT_ARRAY(1017, true, Point[].class),
-  LINE(628, true, Line.class),
-  LINE_ARRAY(629, true, Line[].class),
-  LSEG(601, true, LineSegment.class),
-  LSEG_ARRAY(1018, true, LineSegment[].class),
-  BOX(603, true, Box.class),
-  BOX_ARRAY(1020, true, Box[].class),
-  PATH(602, true, Path.class),
-  PATH_ARRAY(1019, true, Path[].class),
-  POLYGON(604, true, Polygon.class),
-  POLYGON_ARRAY(1027, true, Polygon[].class),
-  CIRCLE(718, true, Circle.class),
-  CIRCLE_ARRAY(719, true, Circle[].class),
-  HSTORE(33670, true, Object.class),
-  OID(26, true, Object.class),
-  OID_ARRAY(1028, true, Object[].class),
-  VOID(2278, true, Object.class),
-  UNKNOWN(705, false, String.class),
-  TS_VECTOR(3614, false, String.class),
-  TS_VECTOR_ARRAY(3643, false, String[].class),
-  TS_QUERY(3615, false,  String.class),
-  TS_QUERY_ARRAY(3645, false,  String[].class);
+  BOOL(16, true, Boolean.class, JDBCType.BIT),
+  BOOL_ARRAY(1000, true, Boolean[].class, JDBCType.BIT),
```

Comment:
I suppose this should be `ARRAY`.

Reply:
so if that's an array, how do you get the array component ? I'm assuming in case of array that we return the component type of the array.

we could add on ColumnDescriptor an isArray or isMultivalued boolean

Reply:
TBH I have never used a SQL array type in my life, so i'm just not sure, but it would not surprise me if there was no good way, other than calling `Array.getBaseType()`.

Reply:
I added a ColumnDescriptor.isArray method to check whether it's an array and the SQL type will be in this case the component type of the array.

Reply:
OK